pkg/apis/componentconfig/helpers.go

```diff
@@ -61,6 +63,41 @@ func (v IPVar) Type() string {
 	return "ip"
 }
 
+type IPPortVar struct {
+	Val *string
+}
+
+func (v IPPortVar) Set(s string) error {
+	ip := s
+	if index := strings.Index(s, ":"); index != -1 {
```

Comment:
how does this handle ipv6?

`[aa:bb:cc::00]:80`

Reply:
Good catch, I sent this out without considering ipv6. Will fix.

Reply:
net.SplitHostPort()

Reply:
@thockin Thanks, that's something I've been looking for.